src/utilities/oas-validator.ts

```diff
@@ -137,67 +142,74 @@ class OasValidator {
       if (expectedType === 'array') {
         // check that the actual object is an array
         if (!Array.isArray(actual)) {
-          throw new TypeMismatchError(path, expectedType, actualType);
+          failures.push(new TypeMismatch(path, expectedType, actualType));
         }
       } else if (actualType !== expectedType) {
         // check that type matches for other types
-        throw new TypeMismatchError(path, expectedType, actualType);
+        failures.push(new TypeMismatch(path, expectedType, actualType));
```

Comment:
Is it worth continuing validation on a type mismatch? If the schema is expecting a string or number but an object is there will we continue trying to validate the object below?

Reply:
good catch. I'll add a test for this and update to return here